### PR TITLE
Fix the filtering problem on color and normals

### PR DESCRIPTION
**Pull request by samcake**
Which was introduced since we generate mips with QImage (pr 9606)

Enable higher quality of filtering through QImage feature for color images in general to avoid the bad filtering we saw in the distance (dev-welcome ground surface)

For normal Maps, the mip generation is not good enough anyway. so we introduced a simple attenuation based on distance in the shader.
THe normal for a fragment is less affected by the normal map in the distance thus avoiding the bad lighting effect due to the normal map contribution in the distance (in dev-play the ground is showing bad patterns in the distance)

## Test Plan
This should fix the problem of filtering of textures in the distance
- in dev-welcome the flour grid should show correctly filtered
- in dev-playa the ground texture should not get darker and shimery in the distance

image
Dev-playa with current Master

image
Dev-playa with this PR

image
Dev-Welcome with Master

image
Dev-Welcome with this PR

### CODE REVIEW

**Comment by cozza13** [comment]
Testing https://github.com/highfidelity/hifi/pull/9647

**Comment by hifi-gustavo** [comment]
Build console output for commit: b9102204ea17644cbe5c9389cdf0a5ba5d202135 available at: https://gist.github.com/79f11f9d26a4196e4950a9da956dcd54<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9647-b910220.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9647-b910220.dmg

File: libraries/render-utils/src/model_lightmap_normal_map.slf [hunk 1]
```diff
@@ -34,7 +34,7 @@ void main(void) {
     <$fetchMaterialTexturesCoord1(matKey, _texCoord1, _SCRIBE_NULL, lightmapVal)$>
 
     vec3 viewNormal;
-    <$tangentToViewSpace(normalTexel, _normal, _tangent, viewNormal)$>
+    <$tangentToViewSpaceLOD(_position, normalTex, _normal, _tangent, viewNormal)$>
```
Comment (jherico) [inline libraries/render-utils/src/model_lightmap_normal_map.slf]: should be `normalTexel`

File: libraries/render-utils/src/model_lightmap_normal_specular_map.slf [hunk 2]
```diff
@@ -34,7 +34,7 @@ void main(void) {
     <$fetchMaterialTexturesCoord1(matKey, _texCoord1, _SCRIBE_NULL, lightmapVal)$>
     
     vec3 viewNormal;
-    <$tangentToViewSpace(normalTexel, _normal, _tangent, viewNormal)$>
+    <$tangentToViewSpaceLOD(_position, normalTex, _normal, _tangent, viewNormal)$>
```
Comment (jherico) [inline libraries/render-utils/src/model_lightmap_normal_specular_map.slf]: should be `normalTexel`

Comment (samcake) [inline libraries/render-utils/src/model_lightmap_normal_specular_map.slf]: thanks

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2da1801fbf2c268d46b175e41d6b7a7eb64b336c available at: https://gist.github.com/79f11f9d26a4196e4950a9da956dcd54<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9647-2da1801.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9647-2da1801.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 13d1642acea011a5143d3d1dca1d9c0149e554e3 available at: https://gist.github.com/79f11f9d26a4196e4950a9da956dcd54<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9647-13d1642.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9647-13d1642.dmg

**Comment by nbq** [comment]
Testing this PR since @samcake asked other testers to test

**Comment by nbq** [comment]
Tester: Coal
System: i7 6700k, GeForce 970, Windows 10

* Tested with dev-download and PR 9647
* On PR 9647 I see that the ground texture does not get darker and shimery in the distance and also the ground texture looks a lot better instead of a poor repeated texture
* On dev-welcome the lights in the hangers in the distance look lit better instead of in the current dev-download, also the grid texture on this pr filters clearly in the distance instead of looking like it turns into a blurry mess in the distance on the dev-download

Textures and lighting look much better to me on this new PR so I give it a:

**PASS**

**Pull request merged**